server/src/main/java/keywhiz/service/resources/admin/SecretsResource.java

```diff
@@ -141,6 +141,29 @@ protected SanitizedSecret retrieveSecret(@Auth User user, String name) {
   }
 
   /**
+   * List all secret names in the given batch
+   *
+   * @excludeParams user
+   * @param idx the first index to select in a list of secrets sorted by creation time
+   * @param num the number of secrets after idx to select in the list of secrets
+   *
+   * @description Returns a list of secret names for this user, limited by the input idx and num
+   * Used by Keywhiz CLI and the web ui.
+   * @responseMessage 200 Found and retrieved Secret(s)
+   */
+  @Path("batch/{idx}-{num}")
```

Comment:
You could have kept the current API to list all secrets and added an optional idx/num (query parameter).

I feel idx and num are poor choices for path parameters.